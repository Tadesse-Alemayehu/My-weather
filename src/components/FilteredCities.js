import React from 'react';
import { Container, Row } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { IoIosArrowBack } from 'react-icons/io';
import PropTypes from 'prop-types';
import Weather from './Weather';
import { applyFilter } from '../Redux/components/navigation/navigation';

function FilteredCities({ filter, cities }) {
  const dispatch = useDispatch();
  const cancelFilter = () => {
    dispatch(applyFilter('All'));
  };
  let filterValue = '';
  if (filter === 'My cities') filterValue = 0;
  else if (filter === 'Near by cities') filterValue = 1;
  else filterValue = 2;
  let noMatchFound = true;
  return (
    <Container>
      <h3>{filter}</h3>
      <Row>
        {cities.map((city) => {
          if (city.nearMe === filterValue) {
            noMatchFound = false;
            return <Weather key={city.id} city={city} />;
          }
          return null;
        })}
      </Row>
      {noMatchFound && <p>Sorry, we could not found match for this category</p>}
      <IoIosArrowBack className="back-icon" onClick={() => cancelFilter()} />
    </Container>
  );
}
FilteredCities.propTypes = {
  filter: PropTypes.func.isRequired,
  cities: PropTypes.func.isRequired,
};

export default FilteredCities;
