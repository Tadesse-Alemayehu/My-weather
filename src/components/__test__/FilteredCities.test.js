import renderer from 'react-test-renderer'
import FilteredCities from '../FilteredCities'
describe('Snapshot the check for the FilteredCities', () => {

  test('should match the snapshot', () => {
    const filteredCitySnap = renderer
      .create(<FilteredCities filter="My cities" cities={[]} />).toJSON();
    expect(filteredCitySnap).toMatchSnapshot();
  })
})