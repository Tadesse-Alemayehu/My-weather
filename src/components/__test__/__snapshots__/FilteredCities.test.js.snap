// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot the check for the FilteredCities should match the snapshot 1`] = `
<div
  className="container"
>
  <h3>
    My cities
  </h3>
  <div
    className="row"
  />
  <p>
    Sorry, we could not found match for this category
  </p>
</div>
`;
